### PR TITLE
Version 2.0.0

### PR DESCRIPTION
**Pull request by DanAndreasson**
This PR does change a lot

1. Upgrade build system and all dev dependencies
2. Add tests with 100% coverage 🎉 
3. Migrate to TypeScript
4. Use named export instead of default export
5. Use `wheel` event instead of `mousewheel` as mousewheel is deprecated

### CODE REVIEW

File: src/test.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  })
+
+  it("handles ctrl + char", async () => {
+    const { findByText } = render(<TestComponent />)
+
+    fireEvent.keyDown(document, { ctrlKey: true, code: "KeyA" })
+    expect(await findByText("Pressed ctrl + a")).toBeInTheDocument()
+  })
+
+  it("handles ctrl + shift + char", async () => {
```
Comment (jimmyhoglund) [inline src/test.tsx]: Everything looks good! Maybe include in this tests though that "Pressed ctrl + a" is NOT included in the document. And then maybe the same for the other 'it("handles...)' tests?

File: src/index.tsx [hunk 2]
```diff
@@ -1,21 +1,30 @@
 import { useEffect } from "react"
 
+export type Shortcut = {
+  keys: string[]
+  onEvent: (event: ShortcutEvent) => void
+  test?: string
```
Comment (jimmyhoglund) [inline src/index.tsx]: What is "test" for?

File: src/test.tsx (hunk 1 above)
Comment (DanAndreasson) [reply src/test.tsx]: Good suggestion. I'll make sure to add at least one test to test the mature propagation

**Comment by DanAndreasson** [comment]
@jimmyhoglund 

> I don't know if it's needed now that we use TypeScript, but maybe you have to update `types.d.ts` so that the intellisense is up to date for when you use useKeyboardShortcuts in a javascript project as well.

`index.d.ts` is the new main typings file and is created automatically by typescript (located in `dist/index.d.ts`

**Pull request merged**